sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -733,6 +735,17 @@ object EliminateSorts extends Rule[LogicalPlan] {
   }
 }
 
+/**
+ * Removes Sort operation if the child is already sorted
+ */
+object RemoveRedundantSorts extends Rule[LogicalPlan] {
+  def apply(plan: LogicalPlan): LogicalPlan = plan transform {
+    case Sort(orders, true, child) if child.outputOrdering.nonEmpty
+        && SortOrder.orderingSatisfies(child.outputOrdering, orders) =>
+      child
+  }
```

Comment:
You might not want to do it in this PR, but you could easily remove another simple kind of redundant sort, e.g.:

`rel.orderBy('a.desc).orderBy('a.asc)`

(and I think that `orderBy` is not stable, so any two consecutive `orderBy` operators are redundant).

Reply:
Yes, you're right. Probably we can do this in other PR. May you open a JIRA for this? Thanks!

Reply:
This is a good follow-up

Reply:
Filed SPARK-23973 for this